io.openems.edge.evcs.goe.chargerhome/src/io/openems/edge/evcs/goe/chargerhome/GoeAPI.java

```diff
@@ -0,0 +1,259 @@
+				}
+			}
+			else {
+				return this.jsonStatus;
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+	
+	/** Limit MaxEnergy for go-e
+	 * 
+	 * See https://github.com/goecharger
+	 * 
+	 * @return JsonObject with new settings
+	 * @throws OpenemsNamedException on error
+	 */
+	public boolean limitMaxEnergy(boolean limit) {
+			
+		try {
+			JsonObject json = new JsonObject();
+			if (!debugMode) {
+				int stp = 0;
+				if (limit) {
+					stp = 2;
+				}				
+				String URL = "http://" + this.IP + "/mqtt?payload=stp=" + Integer.toString(stp);
+				json = this.sendRequest(URL, "PUT");
+				if (json != null) {
+					return true;
+				}
+				else {
+					return false;
+				}
+			}		
+			else {
+				return true;
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
+			return false;
+		}
+	}
+	
+	/** Sets the MaxEnergy in 0.1 kWh for go-e
+	 * 
+	 * See https://github.com/goecharger
+	 * 
+	 * @return JsonObject with new settings
+	 * @throws OpenemsNamedException on error
+	 */
+	public boolean setMaxEnergy(int maxEnergy) {
+			
+		try {
+			JsonObject json = new JsonObject();
+			if (!debugMode) {
+				if (maxEnergy > 0) {
+					this.limitMaxEnergy(true);
+				}
+				else {
+					this.limitMaxEnergy(false);
+				}				
+				String URL = "http://" + this.IP + "/mqtt?payload=dwo=" + Integer.toString(maxEnergy);
+				json = this.sendRequest(URL, "PUT");
+				if (json != null) {
+					return true;
+				}
+				else {
+					return false;
+				}
+			}		
+			else {
+				return true;
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
+			return false;
+		}
+	}
+
+	/**
+	 * Sends a get or set request to the go-e API.
+	 * 
+	 * @param endpoint the REST Api endpoint
+	 * @return a JsonObject or JsonArray
+	 * @throws OpenemsNamedException on error
+	 */
+	private JsonObject sendRequest(String URL, String Requestmethod) throws OpenemsNamedException {
+		try {
+			URL url = new URL(URL);
+			HttpURLConnection con = (HttpURLConnection) url.openConnection();
+			con.setRequestMethod(Requestmethod);
+			con.setConnectTimeout(5000);
+			con.setReadTimeout(5000);
+			int status = con.getResponseCode();
+			String body;
+			try (BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()))) {
+				// Read HTTP response
+				StringBuilder content = new StringBuilder();
+				String line;
+				while ((line = in.readLine()) != null) {
+					content.append(line);
+					content.append(System.lineSeparator());
+				}
+				body = content.toString();
+			}
+			if (status < 300) {
+				// Parse response to JSON
+				return JsonUtils.parseToJsonObject(body);
+			} else {
+				throw new OpenemsException(
+						"Error while reading from go-e API. Response code: " + status + ". " + body);
+			}
+		} catch (OpenemsNamedException | IOException e) {
+			throw new OpenemsException(
+					"Unable to read from go-e API. " + e.getClass().getSimpleName() + ": " + e.getMessage());
+		}
+	}
+	
+}
```

Comment:
A function that reads out the current (total) power demand from OpenEMS and then adjusts the charging power of the charging station depending on how high it is would also be interesting.

In addition, a PV surplus charging function would also be cool. 

Intended would be that when the battery is charged or fed into the grid that this surplus energy (from 3.3 kW surplus possible with Go-E Home Chargers) is passed on to the charging box (eg at 3.3kW surplus a charging current of 6A, at <3.3kW switch off and at >3.3kW the current is calculated via the power in kW : 230V : 3 (since on L1 L2 L3)).

Reply:
To 1:
In my instalation my whole Home Consumes over the Backup Line (continus) from the Goodwe GW 10 Inverter. So i can max. draw around 3.33kW per Phase (10kW in total). So for me it would be great if the Charger automaticly in-/decrease the Charging Power so that the 3.33kW maximum Power per Phase is not exeded. 

E.g. If i have nothing running my Car should Charge with full 10kW but if i cook or smth. like that and draw 2kW on L1 than it should regulate the Charger from 3.33kW per Phase to the maximum of 1.3kW per Phase(so that the 3.33 kW per Phase will not be reached)[i wrote 1.3kW per Phase because the Charger can not async charging per Phase(my knowlege point) so i must limit L1 L2 and L3 when only L1 whould be needed]

To 2:
That' great!

To 3:
The 3.3kW is the minimum Charging Power for a Charger with three Phase's connected.
(Minimum Current per Phase is 6A so 230V * 6A * 3 ≈ 3.3kW)

If you only connected your Charger with one Phase than the minimum Xharging Power is 1.1kW (230V*6A)

Reply:
Ok thanks for explanation.
in 3. your wrong - I think the 3.3 was only for the calculation in point 1. 
230 * 6 * 3 = 4.140    With one phase: 1.380

But the main point for me is that the implementation of the charging station has nothing to do with any storage, pv or e.g. your heat pump.

The EVCS itself is only responsible for reading Data and provide commands to set the charge power.

Even the storage system itself is not doing anything as a standalone Component.

For the entire logic, we are using separate controllers, that expecting e.g. a EVCS and a ESS - no matter what evcs and ess configured.

As long as the given Components provides the complete Nature Channels and their functionality.
https://github.com/OpenEMS/openems/blob/f57602689f46a31e47e779068a5f742c624ec03a/io.openems.edge.controller.evcs/src/io/openems/edge/controller/evcs/EvcsController.java#L252

The evcs controller is calculating the current power, depending on the current user inputs and the current storag & meter informations

Reply:
Yea if you calc the Input-Value's you totaly right its about 4,1kW 
but i said 3,3kW because at my charger if i set Current to 6A only around 5.3A are flowing and 212V (says the Charger itself)
In the real world this values arent true. My Car shows e.g. 3,7kW when the charger say's 3,3kW. This comes from the diffrence the Charger shows in Voltage (Measured Voltage at Charger is ≈ 226V and Charger say's thers only 212V)


Ok thats absolutly clear now but as you sayed the Evcs - Controller does this (is this right or am i wrong?) is it than possible to say this Controller "hey when you over 10kW shut down the charger"?